### PR TITLE
refactor(dev): refactor config handling

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
close https://github.com/risingwavelabs/risingwave/issues/5676

- Some trivial refactors on risedev config expansion for better readability
- Change `risedev.yml` 
```yaml
# old
risedev:
  profile-name:
    - use: ...

# new
risedev:
  profile-name:
    config-path: ... # It will be copied to `.risingwave/config/risingwave.toml` , which will be used for execution.
    steps:
      - use: ...
```
- Refactor config
  1. use a single `RwConfig` instead of multiple `XxConfigs` (I'm not sure why it was organized like that but it's a little bit cumbersome and confusing.)
  2. organize default values for configs
  3. move some system configs from cli options to `risingwave.toml` (only meta has such things)

### CODE REVIEW

File: src/cmd_all/src/playground.rs [hunk 1]
```diff
@@ -26,13 +25,12 @@ use risedev::{
...
 };
+use risingwave_common::config::load_config;
 use tokio::fs::File;
 use tokio::io::AsyncReadExt;
 use tokio::signal;
 
-async fn load_risedev_config(
-    profile: &str,
-) -> Result<(Vec<String>, HashMap<String, ServiceConfig>)> {
+async fn load_risedev_config(profile: &str) -> Result<(Option<String>, Vec<ServiceConfig>)> {
```
Comment (xxchan) [inline src/cmd_all/src/playground.rs:33]: Previously the first return value is not useful. The new first return value is only for playground.

File: src/frontend/src/session.rs [hunk 2]
```diff
@@ -258,13 +258,12 @@ impl FrontendEnv {
     pub async fn init(
         opts: &FrontendOpts,
     ) -> Result<(Self, JoinHandle<()>, JoinHandle<()>, Sender<()>)> {
-        let frontend_config: FrontendConfig = load_config(&opts.config_path).unwrap();
-        let batch_config: BatchConfig = load_config(&opts.config_path).unwrap();
```
Comment (xxchan) [inline src/frontend/src/session.rs:262]: Previously the `batch_config` will load nothing..

File: src/meta/src/config.rs [hunk 3]
```diff
@@ -1,70 +0,0 @@
...
-    #[serde(default)]
-    pub meta: MetaConfig,
-    // Below for streaming.
-    #[serde(default)]
-    pub streaming: StreamingConfig,
-}
-
-#[derive(Clone, Debug, Serialize, Deserialize)]
-#[serde(deny_unknown_fields)]
-pub struct MetaConfig {
```
Comment (xxchan) [inline src/meta/src/config.rs:29]: Moved to `config.rs`

File: src/meta/src/lib.rs [hunk 4]
```diff
@@ -98,61 +97,35 @@ pub struct MetaNodeOpts {
...
-    #[clap(long)]
-    enable_compaction_deterministic: bool,
-
-    /// Enable sanity check when SSTs are committed. By default disabled.
-    #[clap(long)]
-    enable_committed_sst_sanity_check: bool,
-
-    #[clap(long, default_value = "10")]
-    node_num_monitor_interval_sec: u64,
-
```
Comment (xxchan) [inline src/meta/src/lib.rs:136]: Does it look good to put all of these in config file? 🤔

File: src/common/src/config.rs [hunk 5]
```diff
@@ -485,27 +589,3 @@ mod default {
         }
     }
 }
-
-pub mod constant {
```
Comment (xxchan) [inline src/common/src/config.rs:489]: Moved to a different file

**Review by lmatz — APPROVED** [review]
LGTM!

File: src/config/README.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+## Use `risingwave.toml` in `RiseDev`
+
+The config files defined here are mainly used for testing and development purposes. 
+
+`./risingwave.toml` is the default config used by `risedev` .
+
+> **Note**
+>
+> It's value is different from the default values defined in `src/common/src/config.rs` , which is used when the configuration file is not present.
```
Comment (BugenZhao) [inline src/config/README.md:15]:
```suggestion
> Its value is different from the default values defined in `src/common/src/config.rs` , which is used when the configuration file is not present.
```

File: src/config/ci-compaction-test/risingwave.toml [hunk 7]
```diff
@@ -0,0 +1,51 @@
+[meta]
+enable_compaction_deterministic = true
+max_heartbeat_interval_secs = 600
+
+# The following is copied from ../risingwave.toml
```
Comment (BugenZhao) [inline src/config/ci-compaction-test/risingwave.toml:5]: I'm worrying that it'll be hard to maintain in the future. 🥵 Can we introduce some mechanisms to merge/override/patch config files?

Comment (xxchan) [reply src/config/ci-compaction-test/risingwave.toml:5]: oki

Comment (xxchan) [reply src/config/ci-compaction-test/risingwave.toml]: Updated. PTAL

**Review by zwang28 — APPROVED** [review]
LGTM

File: src/config/default.toml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+[meta]
```
Comment (BugenZhao) [inline src/config/default.toml:1]: What does this file for? 👀

Comment (BugenZhao) [reply src/config/default.toml:1]: I see. It's for risedev. Is this the same as the default configuration if no config file is provided? If nope, then it sounds ambiguous. :(

Comment (xxchan) [reply src/config/default.toml:1]: > Do we ensure that this is the same as the default configuration if no config file is provided?

No 🥶 Can you help it?

**Comment by wangrunji0408** [comment]
The main-cron CI task failed after this PR. It seems that `src/config/ci-compaction-test.toml` is missing. 👀

**Pull request merged**